### DISCUSSION TITLE
As a linux user, how do I simply add an existing audiobook folder from my home directory?

### DISCUSSION THREAD

**Discussion by brent-holland** [advplyr/audiobookshelf#3463] [Q&A] [answered]
I am at my wits end.  

So I have my audible library downloaded in /home/mortal/OpenAudible/books 

When I try to add that directory, I get the following error:
```
Failed to ensure folder dir \"/home/mortal/OpenAudible/books\" Error: EACCES: permission denied, mkdir '/home/mortal/OpenAudible/books'","levelName":"ERROR","level":4}
```

I have tried the following:

```
chmod -R 770 /home/mortal/OpenAudible/books 
# I even gave it ownership of the directory, which I really shouldn't have to do
sudo chown -R audiobookshelf:audiobookshelf /home/mortal/OpenAudible/books
>drwxrwx---  2 audiobookshelf audiobookshelf   12K Dec 24  2023 .
```

No matter what, I _still_ cannot add it.  I keep getting the same error.  I shouldn't have to give this app ownership of my audiobooks.  There has to be a better way?

**Comment by advplyr** [comment]
Most likely it's because it is in another users home folder

**Comment by brent-holland** [comment]
It's in my home folder. 'mortal' is my non root account that has all of my existing audiobooks. I don't understand how this very common use case wasn't addressed in the instructions, but I'm happy to open a pr once we solve the problem

**Reply by advplyr** [reply]
The user that the service is set to run as is `audiobookshelf` which wouldn't have access to the home folder of other users by default

**Comment by brent-holland** [comment]
Yes, I understand that. If you read my previous comments you'll see that I tried to give audiobookshelf user access to those folders, but it still failed. What's the proper way to do that here?

**Reply by advplyr** [reply]
https://unix.stackexchange.com/questions/507571/how-to-allow-folder-permission-for-another-user-in-linux

**Reply by brent-holland** [reply, marked as answer]
Yeah I wound up having to do the following

```
sudo groupadd book_readers
sudo usermod -aG book_readers $USERNAME
sudo usermod -aG book_readers audiobookshelf

sudo mkdir -p /shared/books
mv /home/$USERNAME/yourAudiobookFolder /shared/books/ # change me!

sudo chown -R $USERNAME:book_readers shared
chmod -R g+rwX shared
sudo systemctl restart audiobookshelf.service
```
You might want to update the documentation on setting up the library because this was pretty confusing.  I honestly think maybe I had it earlier and didn't do a restart.
